www/js/metrics-factory.js

```diff
@@ -5,37 +5,61 @@ angular.module('emission.main.metrics.factory', ['emission.plugin.kvstore'])
 .factory('FootprintHelper', function() {
   var fh = {};
   var footprint = {
-    train: 92/1609,
-    car: 287/1609,
-    ON_FOOT: 0,
-    BICYCLING: 0
-  }
-  var readable = function(v) {
-    return v > 9999? Math.round(v / 1000) + 'k kg CO₂' : Math.round(v) + ' kg CO₂';
+    ON_FOOT:      0,
+    BICYCLING:    0,
+    CAR:        267/1609,
+    BUS:        278/1609,
+    TRAIN:       92/1609,
+    AIR_OR_HSR: 217/1609
   }
   var mtokm = function(v) {
     return v / 1000;
   }
-  fh.getFootprintRaw = function(distance, mode) {
-    if (mode === "IN_VEHICLE") {
-      return [footprint.train * mtokm(distance), footprint.car * mtokm(distance)];
-    } else {
-      return footprint[mode] * mtokm(distance);
+
+  fh.readableFormat = function(v) {
+    return v > 9999? Math.round(v / 1000) + 'k kg CO₂' : Math.round(v) + ' kg CO₂';
+  }
+  fh.getFootprintForMetrics = function(userMetrics) {
+    var result = 0;
+    for (var i in userMetrics) {
+      var mode = userMetrics[i].key;
+      if (mode in footprint) {
+        result += footprint[mode] * mtokm(userMetrics[i].values);
+      }
+      else if (mode == "IN_VEHICLE") {
+        result += ((footprint[CAR] + footprint[BUS] + footprint[TRAIN]) / 3) * mtokm(userMetrics[i].values);
```

Comment:
Previously you had `IN_VEHICLE` as an average of "train" and "car". Would it make sense now to keep that option and average `TRAIN`, `CAR` and `BUS`, or would you have a different suggestion?

Reply:
average is fine by me